tests/unittests/algo/test_tpe.py

```diff
@@ -0,0 +1,220 @@
+        assert len(gmm_sampler.pdfs) == 12
+
+    def test_sample(self, gmm_sampler):
+        """Test GMMSampler sample function"""
+        points = gmm_sampler.sample(25)
+        points = numpy.array(points)
+
+        assert len(points) <= 25
+        assert numpy.all(points >= -3)
+        assert numpy.all(points < 3)
+
+    def test_get_loglikelis(self, gmm_sampler):
+        """Test to get log likelis of points"""
+        points = numpy.random.random(25)
+        likelis = gmm_sampler.get_loglikelis(points)
+
+        assert len(likelis) == len(points)
+
+
+class TestTPE():
+    """Tests for the algo TPE."""
+
+    def test_seed_rng(self, tpe):
+        """Test that algo is seeded properly"""
+        tpe.seed_rng(1)
+        a = tpe.suggest(1)[0]
+        assert not numpy.allclose(a, tpe.suggest(1)[0])
+
+        tpe.seed_rng(1)
+        assert numpy.allclose(a, tpe.suggest(1)[0])
+
+    def test_set_state(self, tpe):
+        """Test that state is reset properly"""
+        tpe.seed_rng(1)
+        state = tpe.state_dict
+        a = tpe.suggest(1)[0]
+        assert not numpy.allclose(a, tpe.suggest(1)[0])
+
+        tpe.set_state(state)
+        assert numpy.allclose(a, tpe.suggest(1)[0])
+
+    def test_split_trials(self, tpe):
+        """Test observed trails can be split based on TPE gamma"""
+        space = Space()
+        dim1 = Real('yolo1', 'uniform', -3, 6)
+        space.register(dim1)
+
+        tpe.space = space
+        tpe.gamma = 0.25
+
+        points = numpy.linspace(-3, 3, num=10, endpoint=False)
+        results = numpy.linspace(0, 1, num=10, endpoint=False)
+        for point, result in zip(points, results):
+            tpe.observe([[point]], [{'objective': result}])
+
+        below_points, above_points = tpe.split_trials()
+
+        assert below_points == [[-3.0], [-2.4], [-1.8]]
+        assert len(above_points) == 7
+
+        tpe.gamma = 0.2
+        below_points, above_points = tpe.split_trials()
+
+        assert below_points == [[-3.0], [-2.4]]
+        assert len(above_points) == 8
+
+    def test_unsupported_space(self):
+        """Test tpe only work for supported search space"""
+        space = Space()
+        dim = Integer('yolo1', 'uniform', -2, 4)
+        space.register(dim)
+
+        with pytest.raises(ValueError) as ex:
+            TPE(space)
+
+        assert 'TPE now only supports Real Dimension' in str(ex.value)
+
+        space = Space()
+        dim = Real('yolo1', 'norm', 0.9)
+        space.register(dim)
+
+        with pytest.raises(ValueError) as ex:
+            TPE(space)
+
+        assert 'TPE now only supports uniform as prior' in str(ex.value)
+
+        space = Space()
+        dim = Real('yolo1', 'uniform', 0.9, shape=(2, 1))
+        space.register(dim)
+
+        with pytest.raises(ValueError) as ex:
+            TPE(space)
+
+        assert 'TPE now only supports 1D shape' in str(ex.value)
+
+    def test_suggest(self, tpe):
```

Comment:
There should also be a `test_suggest_n_initial_points` and `test_suggest_n_ei_candidates` to make sure they are supported properly. Especially for `n_initial_points`.